app/lib/frontend/static_files.dart

```diff
@@ -173,3 +183,46 @@ Future updateLocalBuiltFiles() async {
     }
   }
 }
+
+Future _updateAdminDartJs() async {
+  final pubAdminDirPath = path.join(_resolveRootPath(), 'pkg', 'pub_admin');
+  final pubAdminBuildJs =
+      new File(path.join(pubAdminDirPath, 'build', 'pub_admin.dart.js'));
+  if (!pubAdminBuildJs.existsSync()) {
+    final pubGetResult = await runProc(
+      'pub',
+      ['get'],
+      workingDirectory: pubAdminDirPath,
+      timeout: const Duration(minutes: 2),
+    );
+    if (pubGetResult.exitCode != 0) {
+      final message = 'Unable to compile pub_admin\n\n'
+          'exitCode: ${pubGetResult.exitCode}\n'
+          'STDOUT:\n${pubGetResult.stdout}\n\n'
+          'STDERR:\n${pubGetResult.stderr}';
+      throw new Exception(message);
+    }
+    final pr = await runProc(
+      'pub',
+      ['run', 'webdev', 'build'],
```

Comment:
I might be new to dart, but why not use `package:build` and run `pub run build_runner build` as part of the commands in the `Dockerfile` ?

Reply:
There is a benefit/tradeoff for both: here, it helps the development, as it will do a build on each startup of the app (local development only needs to care about restarting). In the `Dockerfile` it would make the startup of the instances faster.